pages/api/message/get-admin-message.ts

```diff
@@ -0,0 +1,20 @@
+import adminGetChannelMessage from "databases/functions/ChannelMessages/adminGetChannelMessage";
+import { NextApiRequest, NextApiResponse } from "next";
+
+export default async function getAdminMessage(req: NextApiRequest, res: NextApiResponse) {
+    const {limit = 10, offset = 0} = req.query
+    try {
+        const messages = await adminGetChannelMessage(limit as number, offset as number)
+        return res.status(200).json({
+            success: true,
+            message: 'success',
+            next: parseInt(offset as string) + parseInt(limit as string),
+            data: messages
+        })
+    } catch (error) {
+        return res.status(500).json({
+            success: false,
+            message: 'error',
+        })
+    }
+}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code looks good overall, but there's a potential issue with the type casting of `limit` and `offset` from `req.query`. These are received as strings and you're casting them to numbers directly. This could lead to NaN if the query parameters are not valid numbers. It would be better to validate these inputs before using them.

```diff
    const {limit = 10, offset = 0} = req.query
    try {
+        if (isNaN(Number(limit)) || isNaN(Number(offset))) {
+            return res.status(400).json({
+                success: false,
+                message: 'Invalid limit or offset',
+            })
+        }
        const messages = await adminGetChannelMessage(Number(limit), Number(offset))
```

Also, it's a good practice to handle specific errors in your catch block. This way, you can provide more meaningful error messages and status codes based on the type of error.

```diff
    } catch (error) {
        return res.status(500).json({
+            if (error instanceof SomeSpecificError) {
+                return res.status(400).json({
+                    success: false,
+                    message: 'Some specific error occurred',
+                })
+            }
            success: false,
            message: 'error',
        })
    }
}
```
Replace `SomeSpecificError` with the actual error types you expect could occur during the execution of your code.


<!-- This is an auto-generated comment by OSS CodeRabbit -->